### PR TITLE
Create account

### PR DESCRIPTION
**Pull request by wSedlacek**
# Description
Provides a UI where a user can Signup/Login to an account

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [x] Complete, tested, ready to review and merge
- [x] Complete, but not tested (may need new tests)
- [x] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [x] Backend Testing for Validation
- [x] Backend Testing for Route response
- [x] Frontend Hook Functionality

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Comment by wSedlacek** [comment]
[[Parent] - Create an account](https://trello.com/c/3zKsookg/19-parent-create-an-account)

**Review by ghost — COMMENTED** [review]
readuced budle size. added react-app-rewired

**Review by ghost — COMMENTED** [review]
sign up formatting with material ui and form setting up.

**Review by ghost — COMMENTED** [review]
setting up of sign up and connecting with Back end.

**Review by ghost — COMMENTED** [review]
fixed sign in page to make it functional with on change. and added types as email and password.

**Comment by wSedlacek** [comment]
If this could be reviewed that would be awesome.
Let me know if there are any areas that you would like to see comments.

File: src/app/hooks/form/form.hook.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    const handleBoolChange = (key: string) => (e: React.ChangeEvent<HTMLInputElement>) =>
+        setState({ ...state, [key]: e.target.checked });
+
+    const handleSubmitBuilder = (callback: (state: S) => void) => (
+        e: React.FormEvent<HTMLFormElement>
+    ) => {
+        e.preventDefault();
+        callback(state);
+    };
```
Comment (EricBanker12) [inline src/app/hooks/form/form.hook.ts:47]: This highly nested function is a bit confusing, and a comment would be helpful.

File: src/app/components/sign-up/sign-up.component.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        <>
+            <Typography variant='h3' gutterBottom>
+                Sign Up
+            </Typography>
+            <Typography variant='subtitle2'>Start your child reading stories today!</Typography>
+            <form className={form} onSubmit={handleSubmit}>
+                <TextField
+                    fullWidth
+                    type='email'
+                    label='Username'
```
Comment (EricBanker12) [inline src/app/components/sign-up/sign-up.component.tsx]: Username is an unintuitive label for email.

File: src/app/util/requestFactory.ts [hunk 3]
```diff
@@ -1,10 +1,8 @@
 import axios from 'axios';
 
-const endpoint = '';
-
 export const requestFactory = () =>
     axios.create({
-        baseURL: endpoint,
+        baseURL: process.env.REACT_APP_ENDPOINT,
```
Comment (EricBanker12) [inline src/app/util/requestFactory.ts]: Locally, requests are defaulting to localhost:3000. I believe this is the cause.

**Review by EricBanker12 — CHANGES_REQUESTED** [review]
Signup renders, but submission is using wrong base url locally. More comments would be appreciated for the custom hooks.

**Review by EricBanker12 — APPROVED** [review]
Everything appears to be working, but there is a minor memory leak I believe is caused by form.hook or api.hook on login due to triggering a state change in a hook called by an unmounted component after redirecting to dashboard.

**Pull request merged**